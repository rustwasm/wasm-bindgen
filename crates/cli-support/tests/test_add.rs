const WASM: &[&str] = &[
    std::env!("CARGO_MANIFEST_DIR"),
    "/../../examples/wasm-in-wasm/src/add.wasm",
];

mod common;

test_output_snapshot!(test_bundler_js, false, |b: &mut Bindgen| {
    b.bundler(true).unwrap();
});
test_output_empty!(test_bundler_start, true, |b: &mut Bindgen| {
    b.bundler(true).unwrap();
});
test_output_snapshot!(test_deno_js, false, |b: &mut Bindgen| {
    b.deno(true).unwrap();
});
test_output_empty!(test_deno_start, true, |b: &mut Bindgen| {
    b.deno(true).unwrap();
});
test_output_snapshot!(test_nodejs_js, false, |b: &mut Bindgen| {
    b.nodejs(true).unwrap();
});
test_output_empty!(test_nodejs_start, true, |b: &mut Bindgen| {
    b.nodejs(true).unwrap();
});
test_output_snapshot!(test_nodejs_module_js, false, |b: &mut Bindgen| {
    b.nodejs_module(true).unwrap();
});
test_output_snapshot!(test_nodejs_module_start, true, |b: &mut Bindgen| {
    b.nodejs_module(true).unwrap();
});
test_output_snapshot!(test_web_js, false, |b: &mut Bindgen| {
    b.web(true).unwrap();
});
test_output_empty!(test_web_start, true, |b: &mut Bindgen| {
    b.web(true).unwrap();
});
test_output_snapshot!(test_nomodule_js, false, |b: &mut Bindgen| {
    b.no_modules(true).unwrap();
});
test_output_empty!(test_nomodule_start, true, |b: &mut Bindgen| {
    b.no_modules(true).unwrap();
});
