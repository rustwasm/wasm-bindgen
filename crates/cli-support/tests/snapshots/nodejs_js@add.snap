---
source: crates/cli-support/tests/test_add.rs
expression: output.js()
---
let imports = {};
let wasm;

const path = require('path').join(__dirname, 'add_bg.wasm');
const bytes = require('fs').readFileSync(path);

const wasmModule = new WebAssembly.Module(bytes);
const wasmInstance = new WebAssembly.Instance(wasmModule, imports);
wasm = wasmInstance.exports;
module.exports.__wasm = wasm;
